checkov/terraform/graph_builder/foreach/abstract_handler.py

```diff
@@ -212,67 +149,101 @@ def _update_attributes(self, attrs: dict[str, Any], key_to_val_changes: dict[str
+        return idx_with_separator
 
-        idx_to_change = new_key or new_value
-        self._add_index_to_block_properties(new_resource, idx_to_change)
-        if foreach_idx == 0:
-            self.local_graph.vertices[resource_idx] = new_resource
+    @staticmethod
+    def _update_resolved_entry_for_tf_definition(child: TerraformBlock, original_foreach_or_count_key: int | str,
+                                                 original_module_key: TFModule) -> None:
+        if child.block_type == BlockType.RESOURCE:
+            child_name, child_type = child.name.split('.')
+            config = child.config[child_name][child_type]
         else:
-            self.local_graph.vertices.append(new_resource)
+            config = child.config.get(child.name)
+        if isinstance(config, dict) and config.get(RESOLVED_MODULE_ENTRY_NAME) is not None:
+            tf_moudle: TFModule = config[RESOLVED_MODULE_ENTRY_NAME][0].tf_source_modules
+            ForeachAbstractHandler._update_nested_tf_module_foreach_idx(original_foreach_or_count_key,
+                                                                        original_module_key,
+                                                                        tf_moudle)
+
+    def _handle_static_statement(self, block_index: int, sub_graph: TerraformLocalGraph | None = None) -> \
+            list[str] | dict[str, Any] | int | None:
+        attrs = self.local_graph.vertices[block_index].attributes if not sub_graph \
+            else sub_graph.vertices[block_index].attributes
+        foreach_statement = attrs.get(FOREACH_STRING)
+        count_statement = attrs.get(COUNT_STRING)
+        if foreach_statement:
+            return self._handle_static_foreach_statement(foreach_statement)
+        if count_statement:
+            return self._handle_static_count_statement(count_statement)
+        return None
 
-    def _create_new_resources_foreach(self, statement: list[str] | dict[str, Any], block_idx: int) -> None:
-        main_resource = self.local_graph.vertices[block_idx]
+    def _handle_static_foreach_statement(self, statement: list[str] | dict[str, Any])\
+            -> list[str] | dict[str, Any] | None:
         if isinstance(statement, list):
-            for i, new_value in enumerate(statement):
-                self._create_new_resource(main_resource, new_value, new_key=new_value, resource_idx=block_idx, foreach_idx=i)
-        if isinstance(statement, dict):
-            for i, (new_key, new_value) in enumerate(statement.items()):
-                self._create_new_resource(main_resource, new_value, new_key=new_key, resource_idx=block_idx, foreach_idx=i)
+            statement = self.extract_from_list(statement)
+        evaluated_statement = evaluate_terraform(statement)
+        if isinstance(evaluated_statement, str):
+            try:
+                evaluated_statement = json.loads(evaluated_statement)
+            except ValueError:
+                pass
```

Comment:
do we want to add a log here? maybe in `debug` level so it won't spam in the regular run

Reply:
Currently, we want to get the errors but not fail the flow, when we feel comfortable with this feature we will move it to debug.